src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +35,11 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .map(n -> n % 2 == 1 ? numbers.get(n) - 1 : numbers.get(n))
+                .filter(var -> var % 2 == 1)
+                .average()
+                .orElseThrow();
```

Comment:
[If you need to throw exception when no result found after stream execution sometimes get()/getAsDouble() may help, try to google what does it do.](https://github.com/mate-academy/jv-stream-github-practice/blob/main/checklist.md#try-to-use-a-minimum-amount-of-stream-operations)
```suggestion
                .getAsDouble();
```